File: ruby/pangram/pangram.rb
# frozen_string_literal: true

# Pangram
class Pangram
  ALPHABET = %w|
    a z e r t y u i o p
    q s d f g h j k l m
    w x c v b n
    |
  def self.pangram?(word)
    word = word.downcase
    ALPHABET.all? do |l|
      word.include?(l)
    end
  end
end

File: pangram/pangram.exs
defmodule Pangram do
  @alphabet [
    "a",
    "z",
    "e",
    "r",
    "t",
    "y",
    "u",
    "i",
    "o",
    "p",
    "q",
    "s",
    "d",
    "f",
    "g",
    "h",
    "j",
    "k",
    "l",
    "m",
    "w",
    "x",
    "c",
    "v",
    "b",
    "n"
  ]

  @doc """
  Determines if a word or sentence is a pangram.
  A pangram is a sentence using every letter of the alphabet at least once.

  Returns a boolean.

    ## Examples

      iex> Pangram.pangram?("the quick brown fox jumps over the lazy dog")
      true

  """

  @spec pangram?(String.t()) :: boolean
  def pangram?(sentence) do
    sentence = sentence |> String.downcase
    @alphabet |> Enum.all?(fn x -> sentence |> String.contains?(x) end)
  end
end
